docs: clarify documentation for matchPaths (#19246)
renovatebot/renovate: docs/usage/configuration-options.md

### matchPaths

> Renovate will match `matchPaths` against a minimatch glob pattern.

Reviewer: > Renovate will match `matchPaths` against a minimatch glob pattern.

This sentence has the word `match` in it three times:

- `match`
- `matchPaths`
- `minimatch`

How about this?

```suggestion
Renovate finds the file(s) listed in `matchPaths` with a minimatch glob pattern.
```

> For example the following would match any `package.json`:

Reviewer 2: ```suggestion
For example the following would match any `package.json`, including files like `backend/package.json`:
```

```json

>   "matchPaths": ["package.json"],

Reviewer: ```suggestion
{
  "packageRules": [
    {
      "description": "Find package.json files and do INSERT EXPLANATION",
      "matchPaths": ["package.json"],
    }
  ]
}
```

Author: I'd say this is mostly used to group dependencies somehow, does this help?

```suggestion
"matchPaths": ["package.json"],
{
  "packageRules": [
    {
      "description": "Group dependencies from package.json files",
      "matchPaths": ["**/package.json"],
      "groupName": "All package.json changes "
    }
  ]
}
```

Author: (I also changed `package.json` to `**/package.json` to make clear that this needs to be a pattern)

Reviewer: ```json

```

> While the following would match any file in directories starting with `app-`:

Reviewer 2: ```suggestion
The following would match any file in directories starting with `app-`:
```

```json

>   "matchPaths": ["app-*/**"],

Reviewer: ```suggestion
{
  "packageRules": [
    {
      "description": "Match any file in directories starting with app- and do INSERT EXPLANATION",
      "matchPaths": ["app-*/**"],
    }
  ]
}
```

Author: This is very useful in monorepos where you want to have separate PRs per component. Is this better?

```suggestion
{
  "packageRules": [
    {
      "description": "Group all dependencies from the app directory",
      "matchPaths": ["app/**"],
      "groupName": "App dependencies" 
    }
  ]
}
```

### matchPaths

Renovate finds the file(s) listed in `matchPaths` with a minimatch glob pattern.

For example the following would match any `package.json`, including files like `backend/package.json`:

```json
{
  "packageRules": [
    {
      "description": "Group dependencies from package.json files",
      "matchPaths": ["**/package.json"],
      "groupName": "All package.json changes"
    }
  ]
}
```

> The following would match any file in directories starting with `app-`:

Reviewer 2: ```suggestion
The following would match any file in directories starting with `app/`:
```
